indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Encoding.kt

```diff
@@ -31,8 +31,9 @@ internal fun wrapInUnsafeSource(bytes: ByteArray, startIndex: Int = 0, endIndex:
 internal inline fun throughBuffer(operation: (Buffer) -> Unit): ByteArray =
     Buffer().also(operation).readByteArray()
 
+@OptIn(UnsafeIoApi::class)
 internal inline fun <reified T> ByteArray.throughBuffer(operation: (Source) -> T): T =
-    wrapInUnsafeSource().let { operation(it) }
+    operation(wrapInUnsafeSource())
```

Comment:
stylistic choice ig, but i think `wrapInUnsafeSource().let(operation)` would be more intuitively readable